PowerShell/JumpCloud Commands Gallery/Windows Commands/Windows - Install Remote Assist.md

```diff
@@ -0,0 +1,81 @@
+#### Name
+
+Windows - Install Remote Assist | v1.0 JCCG
+
+#### commandType
+
+windows
+
+#### Command
+
+```
+$installerURL="https://cdn.awsstg.jumpcloud.com/TheJumpCloud/jumpcloud-remote-assist-agent/latest/jumpcloud-assist-app.exe"
+$JumpCloudThumbprint="7A4844FBF481047BEDBB7A8054069C50E449D355"
+$installerTempLocation="C:\Windows\Temp\JumpCloudRemoteAssistInstaller.exe"
+
+Write-Host "Downloading JumpCloud Remote Assist installer now."
+try {
+    $ProgressPreference = "SilentlyContinue"
+    Invoke-WebRequest -Uri $installerURL -OutFile $installerTempLocation
+}
+catch {
+    Write-Error "Unable to download JumpCloud Remote Assist installer."
+    exit 1
+}
+Write-Host "Finished downloading JumpCloud Remote Assist installer."
+
+try {
+    Write-Host "Verifying Authenticode Signature"
+    $authenticode = Get-AuthenticodeSignature "$installerTempLocation"
+    if ( $authenticode.Status -ne "Valid" )
+    {
+        Write-Error "No valid Authenticode signature found, aborting installation"
+        exit 1
+    }
+
+    if ( $authenticode.SignerCertificate.Thumbprint -ne $JumpCloudThumbprint )
+    {
+        Write-Error "No valid Authenticode signature found, aborting installation"
```

Comment:
This error message could be triggered because the code signing certificate was updated after the original expired. This means that the end user must update the thumbprint to the latest code signing certificate. Therefore, I would recommend to change this error message. The message should state that the thumbprint did not match and that the thumbprint might need to get updated to the latest code signing certificate thumbprint to work. We also should document how to get the thumbprint from the executable.

Reply:
If the thumbprint isn't what we expect, then isn't that potentially a supply chain poison? I don't think we want to encourage the admin to work around the invalid signature.